tests/test_rechunk.py

```diff
@@ -119,14 +126,24 @@ def test_rechunk_dataset(
 
     # Validate decoded variables
     dst = xarray.open_zarr(target_store, decode_cf=True)
-    target_chunks_expected = (
-        target_chunks["a"]
-        if isinstance(target_chunks["a"], tuple)
-        else (target_chunks["a"]["x"], target_chunks["a"]["y"])
-    )
+    if chunk_by_dims:
```

Comment:
IMHV having conditionals in tests is usually a good sign that the single test function should be split in two separate ones. This usually helps with readability and makes it easier to understand the problem when the tests break. Obviously you do have some code duplication when you go down this line

Reply:
I did think about that, and generally agree. In order to make that clean I could factor out some of the setup. I originally ended up with tons of duplicated code.

Reply:
I did add an additional test to separate the test cases.